pkg/didcomm/protocol/route/service.go

```diff
@@ -194,14 +216,24 @@ func (s *Service) handleRequest(msg *service.DIDCommMsg, myDID, theirDID string)
 
 func (s *Service) handleGrant(msg *service.DIDCommMsg) error {
 	// unmarshal the payload
-	grant := &Grant{}
+	grantMsg := &Grant{}
 
-	err := json.Unmarshal(msg.Payload, grant)
+	err := json.Unmarshal(msg.Payload, grantMsg)
 	if err != nil {
 		return fmt.Errorf("route grant message unmarshal : %w", err)
 	}
 
-	// TODO https://github.com/hyperledger/aries-framework-go/issues/948 integrate with framework components
+	// check if there are any channels registered for the message ID
+	s.routeRegistrationMapLock.RLock()
+	grantCh := s.routeRegistrationMap[grantMsg.ID]
+	s.routeRegistrationMapLock.RUnlock()
```

Comment:
Move get to different function just to avoid accidents in future

Reply:
[done](https://github.com/hyperledger/aries-framework-go/compare/4d6cf6d2117e03d7fa5fe3e7f5623c5d317dc7df..f2c71755a81cac382569f3cb14fae87828d66b2c#diff-7cc3df019152fe2c557826a312cc0a12R227).